### PR TITLE
improvement(header/server): implement timeout for GetRangeByHeight request

### PR DESCRIPTION
**Pull request by vgonkivs**
Closes https://github.com/celestiaorg/celestia-node/issues/1316

### CODE REVIEW

**Review by distractedm1nd — APPROVED** [review]

**Review by renaynay — APPROVED** [review]

**Comment by renaynay** [comment]
It would be nice to implement a test for this @vgonkivs 🙏🏻

File: header/p2p/server.go [hunk 1]
```diff
@@ -163,7 +163,9 @@ func (serv *ExchangeServer) handleRequest(from, to uint64) ([]*header.ExtendedHe
 		return nil, header.ErrHeadersLimitExceeded
 	}
 	log.Debugw("server: handling headers request", "from", from, "to", to)
-	headersByRange, err := serv.store.GetRangeByHeight(serv.ctx, from, to)
+	ctx, cancel := context.WithTimeout(serv.ctx, time.Minute)
+	defer cancel()
```
Comment (Wondertan) [inline header/p2p/server.go:167]: As we discussed onsite, we don't wanna have our peers _subscribe_ for headers. The minute will end up exactly as a subscription, so I  propose to decrease this to 5 seconds. It's enough time to get the data from the disk, and we don't need it for more

**Review by Wondertan — CHANGES_REQUESTED** [review]

File: header/p2p/server.go (hunk 1 above)
Comment (vgonkivs) [reply header/p2p/server.go:167]: I'm ok with that

**Review by Wondertan — APPROVED** [review]

**Review by distractedm1nd — DISMISSED** [review]

File: header/p2p/server_test.go [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+
+func TestExchangeServer_handleRequestTimeout(t *testing.T) {
+	_, peer := createMocknet(t)
+	s, err := store.NewStore(datastore.NewMapDatastore())
+	assert.NoError(t, err)
+	server := NewExchangeServer(peer, s, "private")
+	err = server.Start(context.Background())
+	assert.NoError(t, err)
+	t.Cleanup(func() {
+		server.Stop(context.Background()) //nolint:errcheck
```
Comment (renaynay) [inline header/p2p/server_test.go:22]: ctx police?

Comment (distractedm1nd) [reply header/p2p/server_test.go:22]: its the lifecycle ctx no?

Comment (vgonkivs) [reply header/p2p/server_test.go:22]: > ctx police?

Not used in Stop

File: header/p2p/server_test.go [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+
+	"github.com/celestiaorg/celestia-node/header/store"
+)
+
+func TestExchangeServer_handleRequestTimeout(t *testing.T) {
+	_, peer := createMocknet(t)
+	s, err := store.NewStore(datastore.NewMapDatastore())
+	assert.NoError(t, err)
```
Comment (renaynay) [inline header/p2p/server_test.go:17]: use require for no errors pls 🙏🏻 unless it's the specific err ur testing for

File: header/p2p/server_test.go [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+	server := NewExchangeServer(peer, s, "private")
+	err = server.Start(context.Background())
+	assert.NoError(t, err)
+	t.Cleanup(func() {
+		server.Stop(context.Background()) //nolint:errcheck
+	})
+
+	_, err = server.handleRequest(1, 200)
```
Comment (renaynay) [inline header/p2p/server_test.go:25]: so will this hang for 1 minute? where does the ctx cancellation happen?

Comment (vgonkivs) [reply header/p2p/server_test.go:25]: https://github.com/celestiaorg/celestia-node/pull/1318#discussion_r1019016683

Comment (vgonkivs) [reply header/p2p/server_test.go:25]: > where does the ctx cancellation happen?

https://github.com/vgonkivs/celestia-node/blob/b06aa8666feb25098d1af582546ff0356d7ee0e4/header/store/heightsub.go#L64

Comment (renaynay) [reply header/p2p/server_test.go:24]: I see now that it's 5 seconds.

**Review by renaynay — APPROVED** [review]

**Review by Wondertan — APPROVED** [review]

**Pull request merged**